<a href="https://colab.research.google.com/github/daviders98/pythonprojects/blob/master/Parcial%20Data%20science%20.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
# PREGUNTA 1
import pandas as pd
location= ("/content/drive/My Drive/Colab Notebooks/datasets/axisdata.csv")
df=pd.read_csv(location)

df["Cars Sold"].mean()


3.9229229229229228

In [ ]:
# PREGUNTA 2
import pandas as pd
location= ("/content/drive/My Drive/Colab Notebooks/datasets/axisdata.csv")
df=pd.read_csv(location)

df["Cars Sold"].max()

7

In [ ]:
# PREGUNTA 3
import pandas as pd
location= ("/content/drive/My Drive/Colab Notebooks/datasets/axisdata.csv")
df=pd.read_csv(location)

df["Cars Sold"].min()

1

In [ ]:
# PREGUNTA 4
import pandas as pd
location= ("/content/drive/My Drive/Colab Notebooks/datasets/axisdata.csv")
df=pd.read_csv(location)
pd.pivot_table(df, index=['Gender'], aggfunc='mean', values=['Cars Sold'])


Cars Sold
Gender           
F        3.760736
M        4.078431

In [ ]:
# PREGUNTA 5
import pandas as pd
location= ("/content/drive/My Drive/Colab Notebooks/datasets/axisdata.csv")
df=pd.read_csv(location)

df2 = df.loc[df['Cars Sold'] > 3]
df2["Hours Worked"].mean()

34.740237691001695

In [ ]:
# PREGUNTA 6
import pandas as pd
location= ("/content/drive/My Drive/Colab Notebooks/datasets/axisdata.csv")
df=pd.read_csv(location)
df["Years Experience"].mean()

3.026026026026026

In [ ]:
# PREGUNTA 7
import pandas as pd
location= ("/content/drive/My Drive/Colab Notebooks/datasets/axisdata.csv")
df=pd.read_csv(location)
df2 = df.loc[df['Cars Sold'] > 3]
df2["Years Experience"].mean()


3.0492359932088284

In [ ]:
# PREGUNTA 8
location= ("/content/drive/My Drive/Colab Notebooks/datasets/axisdata.csv")
df=pd.read_csv(location)

pd.pivot_table(df, index=['SalesTraining'], aggfunc='mean', values=['Cars Sold'], margins='True')

Cars Sold
SalesTraining           
N               3.573850
Y               4.168942
All             3.922923

**Resumen de Indicadores:**
1. Se venden 3.92 carros en promedio por mes
2. Lo máximo de carros que ha vendido un trabajador al mes son 7 
3. Lo mínimo de carros que ha vendido un trabajador al mes es 1
4. Las mujeres venden en promedio 3.7 carros en promedio y los hombres 4.07.
5. Los empleados que venden más de 3 carros al mes, trabajan en promedio 34.7 horas.
6. En promedio los empleados tienen 3.02 años de experiencia
7. En promedio los empleados que venden más de 3 carros al mes, tienen 3.04 años de experiencia
8. Los empleados con capacitación venden 4 carros en promedio al mes, mientras que los que no tienen capacitación, venden 3 carros en promedio al mes. 







In [ ]:
# PREGUNTA 9
import pandas as pd
import statsmodels.formula.api as sm

location= ("/content/drive/My Drive/Colab Notebooks/datasets/axisdata.csv")
df=pd.read_csv(location)

df2=df
co=["name","last","gender","hours","train","exp","sold"]
df2.columns=co 
gendern = pd.get_dummies(df2['gender'])
df2 =df2.join(gendern)
trainn = pd.get_dummies(df2['train'])
df2 =df2.join(trainn)
result =sm.ols(formula="sold ~ F + M + N + Y + hours + exp",data=df2).fit()
result.summary()


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                   sold   R-squared:                       0.110
Model:                            OLS   Adj. R-squared:                  0.107
Method:                 Least Squares   F-statistic:                     30.80
Date:                Thu, 29 Oct 2020   Prob (F-statistic):           3.37e-24
Time:                        02:31:28   Log-Likelihood:                -1781.5
No. Observations:                 999   AIC:                             3573.
Df Residuals:                     994   BIC:                             3598.
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      1.0705      0.107      9.980      0.000       0.860       1.281
F              0.3876      0.071      5.481      0.000       0.249       0.526
M              0.6829      0.070      9.724      0.000       0.545       0.821
N              0.2518      0.072      3.511      0.000       0.111       0.393
Y              0.8187      0.070     11.673      0.000       0.681       0.956
hours          0.0453      0.006      8.156      0.000       0.034       0.056
exp            0.0662      0.033      2.020      0.044       0.002       0.131
==============================================================================
Omnibus:                      951.211   Durbin-Watson:                   2.072
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               63.016
Skew:                          -0.007   Prob(JB):                     2.07e-14
Kurtosis:                       1.770   Cond. No.                     3.05e+16
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The smallest eigenvalue is 1.31e-27. This might indicate that there are
strong multicollinearity problems or that the design matrix is singular.
"""

**Consideramos que el mejor indicador para detectar un buen vendedor es el que vende mas carros con menos horas trabajadas, es decir, una razón de autos/horas, en donde se buscan valores altos o por encima de una meta establecida por la empresa.** 

Para mejorar este indicador, de acuerdo al modelo de regresión lineal multiple, se encontró que las ventas crecen a mayor ritmo si se le brinda un entrenamiento previo al vendedor y además si es del sexo masculino. 